syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.env";
option java_outer_classname = "ReplayBufferProto";
option objc_class_prefix = "HLW";

package replayBuffer;

// The greeting service definition.
service ReplayBuffer {
  rpc ServiceReady (ServiceReadyReq) returns (ServiceReadyResp) {}
  rpc UploadData (UploadDataReq) returns (UploadDataResp) {}
  rpc DownloadData (DownloadDataReq) returns (DownloadDataResp) {}
}

message ServiceReadyReq {
  string requestId = 1;
}

message ServiceReadyResp {
  bool ready = 1;
  string requestId = 2;
}

message Data {
  repeated float values = 1;
  string attributes = 2;  // 包含 shape 和其他信息
}

message UploadDataReq {
  Data data = 1;
  string version = 2;
  string requestId = 3;
}

message UploadDataResp {
  bool done = 1;
  string err = 2;
  string requestId = 3;
}

message DownloadDataReq {
  int32 maxSize = 1;
  string requestId = 2;
}

message DownloadDataResp {
  Data data = 1;
  string err = 2;
  string requestId = 3;
}
